app/services/process_notify_callback.rb

```diff
@@ -1,23 +1,31 @@
 class ProcessNotifyCallback
+  EXPECTED_EMAIL_TYPES = %w[reference_request sign_up_email].freeze
+
   def initialize(notify_reference:, status:)
-    @environment, @email_type, @reference_id = notify_reference.split('-')
+    @environment, @email_type, @id = notify_reference.split('-')
     @status = status
     @not_found = false
   end
 
   def call
-    return unless same_environment? && reference_request_email? && permanent_failure_status?
+    return unless same_environment? && expected_email_type? && permanent_failure_status?
+
+    if reference_request_email?
+      ActiveRecord::Base.transaction do
+        reference = ApplicationReference.find(@id)
 
-    ActiveRecord::Base.transaction do
-      reference = ApplicationReference.find(@reference_id)
+        reference.update!(feedback_status: 'email_bounced')
 
-      reference.update!(feedback_status: 'email_bounced')
+        SendNewRefereeRequestEmail.call(
+          application_form: reference.application_form,
+          reference: reference,
+          reason: :email_bounced,
+        )
+      end
+    else
```

Comment:
Maybe instead `elsif sign_up_email?` and then another else that raises "Not implemented" or something to that effect?

Reply:
I don't think this is needed because of the guard statement particularly the `expected_email_type?` as this checks that the email type provided in the Notify reference matches one of the ones we expect and currently they are only two so it's going to be either a reference request email or a sign up email.

Reply:
Updated to be more explicit.